_studio/mfx_lib/encode_hw/hevc/agnostic/g12/hevcehw_g12_scc.cpp

```diff
@@ -0,0 +1,346 @@
+{
+    if (id != SCC_EXT_ID)
+        return false;
+
+    SccSpsExt& sps = SpsExt::GetOrConstruct(strg, SccSpsExt{});
+
+    sps.curr_pic_ref_enabled_flag = bs.GetBit();
+    sps.palette_mode_enabled_flag = bs.GetBit();
+
+    if (sps.palette_mode_enabled_flag)
+    {
+        sps.palette_max_size = bs.GetUE();
+        sps.delta_palette_max_predictor_size = bs.GetUE();
+        sps.palette_predictor_initializer_present_flag = bs.GetBit();
+
+        if (sps.palette_predictor_initializer_present_flag)
+        {
+            return false;
+        }
+    }
+
+    sps.motion_vector_resolution_control_idc = bs.GetBits(2);
+    sps.intra_boundary_filtering_disabled_flag = bs.GetBit();
+    sps.scc_extension_flag = 1;
+
+    return true;
+}
+
+bool SCC::ReadPpsExt(StorageRW& strg, const Base::PPS&, mfxU8 id, Base::IBsReader& bs)
+{
+    if (id != SCC_EXT_ID)
+        return false;
+
+    auto pExt = make_storable<SccPpsExt>(SccPpsExt{});
+    SccPpsExt& pps = *pExt;
+
+    pps.curr_pic_ref_enabled_flag = bs.GetBit();
+    pps.residual_adaptive_colour_transform_enabled_flag = bs.GetBit();  // MBZ for Gen12
+
+    if (pps.residual_adaptive_colour_transform_enabled_flag)
+    {
+        return false;
+    }
+    pps.palette_predictor_initializer_present_flag = bs.GetBit();   // MBZ for Gen12
+
+    if (pps.palette_predictor_initializer_present_flag)
+    {
+        return false;
+    }
+
+    pps.scc_extension_flag = 1;
+
+    strg.Insert(PpsExt::Key, std::move(pExt));
+
+    return true;
+}
+
+bool SCC::PackSpsExt(StorageRW& strg, const Base::SPS&, mfxU8 id, Base::IBsWriter& bs)
+{
+    if (id != SCC_EXT_ID)
+        return false;
+
+    auto& sps = SpsExt::Get(strg);
+
+    bs.PutBit(sps.curr_pic_ref_enabled_flag);
+    bs.PutBit(sps.palette_mode_enabled_flag);
+
+    if (sps.palette_mode_enabled_flag)
+    {
+        bs.PutUE(sps.palette_max_size);
+        bs.PutUE(sps.delta_palette_max_predictor_size);
+        bs.PutBit(0); // Gen12: palette_predictor_initializer_present_flag - MBZ
+    }
+
+    bs.PutBits(2, 0); // Gen12: motion_vector_resolution_control_idc - MBZ
+    bs.PutBit(0); // Gen12: intra_boundary_filtering_disabled_flag - MBZ
+
+    return true;
+}
+
+bool SCC::PackPpsExt(StorageRW& strg, const Base::PPS&, mfxU8 id, Base::IBsWriter& bs)
+{
+    if (id != SCC_EXT_ID)
+        return false;
+
+    auto& pps = PpsExt::Get(strg);
+
+    bs.PutBit(pps.curr_pic_ref_enabled_flag);
+    bs.PutBit(0); // Gen12: pps.residual_adaptive_colour_transform_enabled_flag - MBZ
+    bs.PutBit(0); // Gen12: pps.palette_predictor_initializer_present_flag - MBZ
+
+    return true;
+}
+
+void SCC::Query1NoCaps(const FeatureBlocks& /*blocks*/, TPushQ1 Push)
+{
+    Push(BLK_SetCallChains
+        , [this](const mfxVideoParam& par, mfxVideoParam&, StorageRW& strg) -> mfxStatus
+    {
+        MFX_CHECK(par.mfx.CodecProfile == MFX_PROFILE_HEVC_SCC, MFX_ERR_NONE);
+        MFX_CHECK(!IsOff(par.mfx.LowPower), MFX_ERR_NONE);
```

Comment:
Update Checking Logic here. If -qsv ff isn't set by user , lib throws unsupported with scc profile in Query.